test/DecryptionFailureTracker-test.ts

```diff
@@ -306,20 +309,16 @@ describe('DecryptionFailureTracker', function() {
 
         tracker.trackFailures();
 
-        expect(counts['OlmUnspecifiedError'])
-            .toBe(3, 'should track three OlmUnspecifiedError, got ' + counts['OlmUnspecifiedError']);
+        expect(trackEventSpy).toHaveBeenCalledTimes(1);
+        expect(trackEventSpy).toHaveBeenCalledWith('E2E', 'Decryption failure', 'UnknownError', '3');
     });
 
     it('should remap error codes correctly', () => {
-        const counts = {};
-        const tracker = new DecryptionFailureTracker(
-            (total, errorCode) => counts[errorCode] = (counts[errorCode] || 0) + total,
-            (errorCode) => Array.from(errorCode).reverse().join(''),
```

Comment:
this seemed important to the test?

Reply:
This was testing that error codes are being run through this map function by asserting the reversed error code was present on the error map `expect(counts['1_EDOC_RORRE']).toBe(1)`

Because types, and I think because it tests the code in use in our app better, I've tested using the actual error code mapper that the singleton instance uses (https://github.com/matrix-org/matrix-react-sdk/blob/113d69c089dd57ece5a1ba621c643e8980512618/src/DecryptionFailureTracker.ts#L48-L60)
and asserted that `MEGOLM_UNKNOWN_INBOUND_SESSION_ID` maps to `OlmKeysNotSentError ` and that the track function is called with the mapped error code.

Reply:
Reverted in favour of #8272